### PR TITLE
Stripe SEPA integration

### LINKED ISSUE #4313: feat: allow stripe to accept SEPA Direct Debit
# Feature Request

## Description
Integration of SEPA payment method from Stripe: https://stripe.com/docs/payments/sepa-debit
This is a widely used payment method in Europe.

## Checklist:
- [ ] I can donate with SEPA Direct Debit in Stripe.
- [ ] `2%` fee include and exclude depends upon pro or free version of Stripe

### PR DESCRIPTION
**Pull request by ravinderk**
## Description
This PR will resolve #4313

## How Has This Been Tested?
Locally

## Types of changes

New feature (non-breaking change which adds functionality)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

### CODE REVIEW

File: includes/gateways/stripe/includes/payment-methods/class-give-stripe-sepa.php [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+
+					// Save donation summary to donation.
+					give_update_meta( $donation_id, '_give_stripe_donation_summary', $donation_summary );
+
+					/**
+					 * This filter hook is used to update the payment intent arguments.
+					 *
+					 * @since 2.5.0
+					 */
+					$intent_args = apply_filters(
```
Comment (DevinWalker) [inline includes/gateways/stripe/includes/payment-methods/class-give-stripe-sepa.php]: @mehul0810 where are we adding / removing the fee based on whether the user is on Stripe free or Premium?

**Comment by DevinWalker** [comment]
## PR Review

1. When SEPA is not the default gateway I'm seeing this error in console: `Uncaught TypeError: Cannot read property 'getAttribute' of null`



This is blocking the rendering the IBAN field:



2. Change the "Hide Icon" option to "Display Icon" and have it enabled by default. This is better UX and less confusing.

3. Change the description of the "Mandate Acceptance Text" to: 

> This text displays below the IBAN field and provides clarity to your donors on how this payment option works. If you would not like this text to display then you can remove the default text. 

4. If I remove the text within the "Mandate Acceptance Text" text and it is a blank field then do not output any text or markup for this field on the donation form. Currently the default text still appears after removing the text and saving this field:

**Comment by mehul0810** [comment]
## PR Review

> 1. When SEPA is not the default gateway I'm seeing this error in console: `Uncaught TypeError: Cannot read property 'getAttribute' of null`
- Fixed this issue and tried with different scenarios switching the payment gateways.

> 2. Change the "Hide Icon" option to "Display Icon" and have it enabled by default. This is better UX and less confusing.
- Fixed and updated the functioning.

> 3. Change the description of the "Mandate Acceptance Text" to: 
> This text displays below the IBAN field and provides clarity to your donors on how this payment option works. If you would not like this text to display then you can remove the default text. 
- Update text as mentioned above.

> 4. If I remove the text within the "Mandate Acceptance Text" text and it is a blank field then do not output any text or markup for this field on the donation form. Currently, the default text still appears after removing the text and saving this field:
- This is hard to achieve as an empty value for admin fields doesn't work with `give_get_option` which means either we can make it work with empty value or default value and it can't work for both. So, the alternative would be to add additional fields with name like "Display Mandate Acceptance Text" and when enabled show the default text and keep the field default enabled. Let me know if you want me to proceed in this way.

**Comment by DevinWalker** [comment]
@mehul0810 I have taken care of 3 and 4. When testing SEPA as the default gateway I'm noticing the icon isn't displaying on page load now or when switching gateways:



When it's not the default gateway it displays fine.

**Comment by mehul0810** [comment]
@DevinWalker I think you have the previous Hide Icon settings set to `disabled` which is causing the icon to hide on IBAN element because when I tested with the new settings named "Display Icon" is it showing properly when the user is first time Give user and also when there is default gateway set as SEPA and the default gateway is set as some other gateway. 

Please try again with a fresh install or changing the settings. Let me know if it still doesn't work, we can do a quick call tomorrow morning.

This is how it looks for first time users and default gateway set to SEPA Direct Debit.
image

**Pull request merged**